test/failure_detector_tests.ml

```diff
@@ -0,0 +1,49 @@
+open Commons
+open Pollinate.Node
+open Lwt.Infix
+
+module SUT = Pollinate.Node.Failure_detector
+
+let node_a =
+  Lwt_main.run
+    (Node.init ~router:Commons.router ~state:["test1"]
+       ~msg_handler:Commons.msg_handler ("127.0.0.1", 3003))
+let node_b =
+  Lwt_main.run
+    (Node.init ~router:Commons.router ~state:["test1"]
+       ~msg_handler:Commons.msg_handler ("127.0.0.1", 3004))
+
+let peer_b = Client.peer_from !node_b
+
+let suspicion_detection () =
+  let open Common.Peer in
+  let _ = add_neighbor (Pollinate.Node.Client.peer_from !node_a) peer_b in
```

Comment:
This works but suspiciously so. It only works because when you construct a peer from a client, the underlying reference to the neighbors/peers hash table is the same. Thus, any change to the neighbors table in the result of `peer_from !node_a` will affect `node_a`'s peer table. I'm not sure we should do this, especially if we will eventually change to using maps which may have different functionality. Instead, let's add a function to the client for adding peers.

Reply:
Oh boy! That's a great catch!
I added a `add_peer` function on Client.ml, but should we still propose the `add_neighbor`?